crates/shared/src/price_estimation/factory.rs

```diff
@@ -273,11 +273,11 @@ impl<'a> PriceEstimatorFactory<'a> {
                             )
                         },
                     )
-                    .find(|external| external.0 == estimator.name())
+                    .find(|external| external.0 == estimator)
```

Comment:
If you configure a driver based estimator called `Baseline` this code will return the built-in version instead of the driver version due the order in which arguments get passed to `PriceEstimationSource`.
I think there are reasons for preferring built-in or user defined and there are ways to work around the name shadowing. Just wanted to point it out in case you consider this a foot gun.
But in any case this behavior might have to be document given the fact that we are in the process of replacing built-in solvers with equivalent colocated solvers.

Reply:
Added a comment to the argument doc string